### PR TITLE
Dual license the BPF part of the codebase as Apache/GPL.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

This allows for headers to be imported into Felix without accidentally picking up GPL requirements.
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
Calico's BPF programs are now dual Apache/GPL licensed to facilitate code sharing with Felix.
```

### CODE REVIEW

**Review by neiljerram — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
